x-pack/solutions/security/plugins/security_solution/docs/rfcs/detection_response/customized_rule_alert_telemetry.md

```diff
@@ -0,0 +1,273 @@
+
+```js
+'kibana.alert.rule.parameters': {
+  rule_source: true,
+}
+```
+
+### Examples
+
+**Example rule object with new field**
+
+```js
+{
+    ...
+    "rule_source": {
+        "type": "external",
+        "is_customized": true,
+        // New field
+        "customized_fields": [
+            { "field_name": 'query', "is_functional": true },
+            { "field_name": 'note', "is_functional": false },
+            { "field_name": 'tags', "is_functional": false },
+        ]
+    },
+    ...
+}
+```
+
+**Example KQL queries in the alerts telemetry cluster**
+
+```
+not kibana.alert.rule.params.rule_customizations.is_functional : true
+```
+A query we could use to filter out any alerts generated by prebuilt rules that have functional fields customized.
+
+
+### Pros  
+- Would have a negligible effect on performance
+  - We already calculate the rule diff everywhere we'd modify this field, would just need to carry over the field names.
+  - No need to run the rule diff calculation at rule execution time, just copy it to the alert object like we do any other field
+- Easy to query on the telemetry side
+  - Building aggregation visualizations and dashboards in the alert telemetry cluster would be fairly straightforward with this implementation. As shown in some of the examples above, we can be granular and specific with our queries.
+
+### Cons 
+- Once we add it there's (basically) no going back
+  - Not necessarily a con but something to consider in terms of longevity 
+
+### Complexity & Cost
+
+This effort would involve:
+ -  adding a new field to our rule schemas
+ -  updating all dependent schemas
+ -  adding the logic to update this field all the same places we update `rule_source`
+ -  writing the new field to the alert documents during rule execution
+ -  writing tests to cover the new field and resulting logic additions
+
+This is all in addition to the then-unblocked telemetry work that could take place once this was implemented.
+
+---
+
+## Appendix
+
+### Defining functional rule fields
+
+| Rule fields                           | Functional field? |
+|---------------------------------------|-------------------|
+| `name`                                | no                |
+| `description`                         | no                |
+| `risk_score`                          | yes               |
+| `severity`                            | yes               |
+| `rule_name_override`                  | yes               |
+| `timestamp_override`                  | yes               |
```

Comment:
```suggestion
| `timestamp_override`                  | no               |
```

? Does this actually change the functionality or does it just modify the timestamp field within the generated alert.